grails-app/controllers/org/bbop/apollo/UserController.groovy

```diff
@@ -479,7 +479,21 @@ class UserController {
                 render status: HttpStatus.UNAUTHORIZED
                 return
             }
-            User user = User.findById(dataObject.userId)
+            User user = null
+            if (dataObject.has('userId')) {
+                user = User.findById(dataObject.userId)
+            }
+            // to support the webservice
+            if (!user && dataObject.has("email")) {
+                user = User.findByUsername(dataObject.email)
```

Comment:
You want to leave this as-is.     You won't be able to change the username / email otherwise.